generator/sponsored_object_storage.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include "platform/platform.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+#include "geometry/latlon.hpp"
+
+#include "base/logging.hpp"
+
+#include <fstream>
+#include <functional>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "boost/geometry.hpp"
+#include "boost/geometry/geometries/box.hpp"
+#include "boost/geometry/geometries/point.hpp"
+#include "boost/geometry/index/rtree.hpp"
+
+namespace generator
+{
+template <typename Object>
+class SponsoredObjectStorage
+{
+public:
+  using ObjectId = typename Object::ObjectId;
+  using ObjectsContainer = std::map<ObjectId, Object>;
+  using FillObject = std::function<void(ObjectsContainer & objects)>;
+
+  SponsoredObjectStorage(double distanceLimitMeters, size_t maxSelectedElements)
+    : m_distanceLimitMeters(distanceLimitMeters)
+    , m_maxSelectedElements(maxSelectedElements)
+  {
+  }
+
+  double GetDistanceLimitInMeters() const
+  {
+    return m_distanceLimitMeters;
+  }
+
+  size_t GetMaxSelectedElements() const
+  {
+    return m_maxSelectedElements;
+  }
+
+  ObjectsContainer const & GetObjects() const
+  {
+    return m_objects;
+  }
+
+  size_t Size() const
+  {
+    return m_objects.size();
+  }
+
+  void SetFillObjects(FillObject const & fn)
+  {
+    m_fillObject = fn;
+  }
+
+  void LoadData(std::string const & dataPath, std::string const & addressReferencePath)
+  {
+    if (dataPath.empty())
+      return;
+
+    std::ifstream dataSource(dataPath);
+    if (!dataSource.is_open())
+    {
+      LOG(LERROR, ("Error while opening", dataPath, ":", strerror(errno)));
+      return;
+    }
+
+    LoadData(dataSource, addressReferencePath);
+  }
+
+  void LoadData(std::istream & src, std::string const & addressReferencePath)
+  {
+    m_objects.clear();
+    m_rtree.clear();
+
+    for (std::string line; std::getline(src, line);)
+    {
+      Object object(line);
+      m_objects.emplace(object.m_id, object);
+    }
+
+    // Try to get object address from existing MWMs.
+    if (!addressReferencePath.empty())
+    {
+      LOG(LINFO, ("Reference addresses for sponsored objects", addressReferencePath));
+      Platform & platform = GetPlatform();
+      std::string const backupPath = platform.WritableDir();
+
+      // MWMs can be loaded only from a writebledir or from a resourcedir,
```

Comment:
Okay. Looks like it's better to remove ForTests suffixes, as these functions may be used in tools too. Feel free to do that in an individual PR.